### DISCUSSION TITLE
image gallery caption options

### DISCUSSION THREAD

**Discussion by shawngiese** [GetPublii/Publii#1823] [Help]
Any way to get the captions to show below the images in a gallery like in the preview? When we publish and view the captions are changed to white and only visible when hovering over the image. Many of our images are white and this makes the published captions hard to read. Also, some of the images are larger when viewed in HTML than when seen in preview. IE, more gallery options would be great. 

Also, was there a UI way to change things like the theme background color? I know we can do it in CSS but I thought the basic colors would be more accessible for each theme.

**Comment by shawngiese** [comment]
We are probably going to stick with the following CSS for now.

```
.gallery__item figcaption {
  color: black !important;
  bottom: 10px;
  opacity: 100 !important;
}
```

**Comment by candidexmedia** [comment]
Can you share some screenshots or screen recordings or what's happening, and let us know which theme?

**Reply by shawngiese** [reply]
Will try and get some screenshots:

- caption hard to read on busy image
- publii preview not what is shown on web preview
- some images show up super big or super tall
- some images are resized in web... like what was nice and cropped in publii preview gets all a new viewing size in web
- long captions cover the image in web (not in publii preview)
 Using Simple 3.0.1.0

**Reply by shawngiese** [reply]
![image](https://github.com/user-attachments/assets/89c72bf9-3708-4dea-bbfc-59e7268a9b64)

**Reply by shawngiese** [reply]
For now we will just use the table layout. We can easily control diplay there with css. For now the image gallerry is too hard to customize for what we want.... Argh, the image size in the publii does not work in tables... will try to force everything in CSS.

In the end css flexbox was our friend. We lose the Publii preview but now columns stack in smaller devices and captions are fixed. 
![image](https://github.com/user-attachments/assets/eb83629b-f5b6-4959-8217-b6b402ecf5bc)

**Reply by candidexmedia** [reply]
Hmmm... Publii uses [Photoswipe](https://photoswipe.com/) for galleries, which might explain why it looks different in your WYSIWYG editor vs the preview.

Give the masonry layout a try, maybe? : https://github.com/GetPublii/Publii/discussions/1387

I'd also consult the Photoswipe docs to get a better sense of how to adjust how captions are displayed. The post editors aren't meant to be a 1 for 1 replica of your actual page.

**Reply by shawngiese** [reply]
Checked out the Photoswipe docs and got lost there. Really liked your image loading mechanism though for the image gallery... so wrote a plugin to overwrite the Photoswipe CSS with simplified CSS to do what we want. If anyone wants to try it out let me know... a work in progress.

**Reply by candidexmedia** [reply]
@shawngiese Awesome!
